inputremapper/daemon.py

```diff
@@ -437,7 +437,7 @@ def start_injecting(self, group_key, preset):
         preset_path = os.path.join(
             self.config_dir,
             "presets",
-            group.name,
+            sanitize_path_component(group.name),
             f"{preset}.json",
         )
```

Comment:
can you test if 
```python
        preset_path = get_preset_path(group.name, preset)
```
works here instead? 
`get_preset_path` is in `inputremapper.configs.paths`

Reply:
This way it does the sanitization correctly, but now it fails to load the profile because it tries to locate it in the root user's home dir, so no joy.

It looks as if the unpatched/unsanitized way uses `daemon.config_dir`, which correctly reports the non-root user, while `get_preset_path` uses `CONFIG_PATH`, which reports `root` (via `pwd.getpwnam(user).pw_dir`). Maybe pass the config dir as an arg to `get_preset_path` so both daemon and client(s) can use it? I don't want to break anything, happy to look into it but it may take a while to study the code further.

Reply:
You are right, the service is running as root. Which is why we didn't use `get_preset_path`. I think your solution is fine as is.